src/snowflake/snowpark/_internal/compiler/utils.py

```diff
@@ -266,13 +269,27 @@ def update_resolvable_node(
         assert node.snowflake_plan is not None
         update_resolvable_node(node.snowflake_plan, query_generator)
         node.analyzer = query_generator
+        node.expr_to_alias.update(node._snowflake_plan.expr_to_alias)
+        node.df_aliased_col_name_to_real_col_name.update(
+            node._snowflake_plan.df_aliased_col_name_to_real_col_name
+        )
```

Comment:
`SelectTableFunction` does not have these properties, right? can this be moved below with _query_params update

Reply:
is this the reason it failed before?

Reply:
no, it actually is not, the reason that fails, is because we didn't set the node._projection_in_str = None,  this part is more to complete the property propogation. 

I think we will need to get some test that requires expr_to_alias and df_aliased_col_name_to_real_col_name, so far i don't think we have any test test that, i have created a ticket here https://snowflakecomputing.atlassian.net/browse/SNOW-1641640

Reply:
@sfc-gh-aalam yes, moved